libs/ui/packages/react/src/components/layout/Carousel/index.tsx

```diff
@@ -94,3 +146,37 @@ const Carousel = ({ children, variant = "default", onChange }: Props) => {
 };
 
 export default Carousel;
+
+const AutoplayFlags = {
+  play: true,
+  stopOnMouseEnter: true,
+  stopOnInteraction: false,
+};
+
+function watchDragX(emblaApi: EmblaCarouselType) {
+  emblaApi.on("pointerDown", watchMouse);
+
+  let start: number | undefined;
+  let end: number | undefined;
+
+  return {
+    get value() {
+      return typeof start === "undefined" || typeof end === "undefined" ? 0 : end - start;
+    },
+    clean: () => {
+      emblaApi.off("pointerDown", watchMouse);
```

Comment:
i don't have insights about the`.off` function implementation, but i'm not very convinced that it will remove watchMouse internal event listener, i think it will add it instead.
To play safe, it's better to create a function that uses an explicit `.removeEventListener(...)` of the `mousemove` event type .

Reply:
Nice catch. I forgot to update the cleanup function after changing the watch logic